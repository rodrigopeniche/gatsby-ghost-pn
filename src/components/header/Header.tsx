import * as React from "react"
import Buttons from "./buttons-section/Buttons"
import HeaderContainer from "./HeaderContainer"
import Label from "./Label"
import HeaderImage from "./HeaderImage"

const Header = ({
    missionStatement = false,
    stateKey
}: {
    missionStatement?: boolean
    stateKey: string
}) => (
    <HeaderContainer missionStatement={missionStatement}>
        <Label stateKey={stateKey}/>
        <HeaderImage />
        <Buttons />
    </HeaderContainer>
)

export default Header
