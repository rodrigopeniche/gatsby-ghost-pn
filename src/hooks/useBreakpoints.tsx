import * as React from "react"
import { useState, useEffect, createContext, useContext } from "react"

const defaultValue = {}

const BreakpointContext = createContext(defaultValue)

const BreakpointProvider = ({
    children,
    queries,
}: {
    children: React.ReactNode | React.ReactNode[]
    queries: { [key: string]: string }
}) => {
    const [queryMatch, setQueryMatch] = useState({})

    useEffect(() => {
        const mediaQueryLists: { [key: string]: any } = {}
        const keys = Object.keys(queries)
        let isAttached = false

        const handleQueryListener = () => {
            const updatedMatches = keys.reduce(
                (acc: { [key: string]: any }, media) => {
                    acc[media] = !!(
                        mediaQueryLists[media] && mediaQueryLists[media].matches
                    )
                    return acc
                },
                {}
            )
            setQueryMatch(updatedMatches)
        }

        if (window && window.matchMedia) {
            const matches: { [key: string]: any } = {}
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media] = window.matchMedia(queries[media])
                    matches[media] = mediaQueryLists[media].matches
                } else {
                    matches[media] = false
                }
            })
            setQueryMatch(matches)
            isAttached = true
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media].addListener(handleQueryListener)
                }
            })
        }

        return () => {
            if (isAttached) {
                keys.forEach(media => {
                    if (typeof queries[media] === "string") {
                        mediaQueryLists[media].removeListener(
                            handleQueryListener
                        )
                    }
                })
            }
        }
    }, [queries])

    return (
        <BreakpointContext.Provider value={queryMatch}>
            {children}
        </BreakpointContext.Provider>
    )
}

function useBreakpoint(): { [key: string]: boolean } {
    const context = useContext(BreakpointContext)
    if (context === defaultValue) {
        throw new Error("useBreakpoint must be used within BreakpointProvider")
    }
    return context
}
export { useBreakpoint, BreakpointProvider }
