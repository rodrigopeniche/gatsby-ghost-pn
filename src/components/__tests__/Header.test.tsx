import * as React from "react"
import * as renderer from "react-test-renderer"
import Header from "../header/Header"
import { BreakpointProvider } from "../../hooks/useBreakpoints"

describe("Article", () => {
    it("renders correctly", () => {
        const tree = renderer.create(
            <BreakpointProvider queries={{}}>
                <Header />
            </BreakpointProvider>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
