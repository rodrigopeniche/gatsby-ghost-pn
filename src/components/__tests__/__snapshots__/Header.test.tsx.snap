// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "linear-gradient(0deg, rgba(196, 196, 196, 0) 0.58%, rgba(196, 196, 196, 0.3) 18px, rgba(196, 196, 196, 0) 19px",
      "boxSizing": "content-box",
      "display": "flex",
      "gridArea": "header",
      "height": "36px",
      "justifyContent": "space-between",
      "margin": "-5px -5px 0",
      "minWidth": "200px",
      "padding": "0 12px 0 16px",
      "position": "relative",
      "textTransform": "uppercase",
    }
  }
>
  <div
    style={
      Object {
        "color": "#91A6BC",
        "fontSize": "16px",
        "fontStyle": "normal",
        "fontWeight": "normal",
        "lineHeight": "18px",
        "textAlign": "center",
      }
    }
  >
    NY phone number lookups
  </div>
  <div
    style={
      Object {
        "height": "130px",
        "left": "calc((100% - 174px)/2)",
        "marginTop": "-15px",
        "position": "absolute",
        "top": "calc((100% - 130px)/2)",
        "width": "174px",
      }
    }
  >
    <img
      style={
        Object {
          "marginBottom": "0px",
          "marginTop": "15px",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginBottom": "5px",
          "marginLeft": "0px",
          "marginTop": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#ee0017",
            "borderRadius": "50%",
            "content": "",
            "display": "inline-block",
            "height": "5px",
            "marginRight": "5px",
            "width": "5px",
          }
        }
      />
      <a
        href="http://members.staterecords.org/customer/login"
        style={
          Object {
            "color": "#305983",
            "fontSize": "12px",
            "lineHeight": "14px",
            "marginLeft": "auto",
            "textDecoration": "none",
          }
        }
      >
        Login
      </a>
    </div>
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginBottom": "5px",
          "marginLeft": "15px",
          "marginTop": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#ee0017",
            "borderRadius": "50%",
            "content": "",
            "display": "inline-block",
            "height": "5px",
            "marginRight": "5px",
            "width": "5px",
          }
        }
      />
      <a
        href="http://members.staterecords.org/customer/help"
        style={
          Object {
            "color": "#305983",
            "fontSize": "12px",
            "lineHeight": "14px",
            "marginLeft": "auto",
            "textDecoration": "none",
          }
        }
      >
        Contact us
      </a>
    </div>
  </div>
</div>
`;
